### ISSUE TITLE
Use hts_open

### ISSUE THREAD

**Issue by jenniferliddle** [wtsi-npg/bambi#44]
Replace open() with hts_open() throughout, so that we can open files directly from irods or AWS or Google web services.

This will involve replacing the existing XML library used by i2b

**Comment by dkj** [comment]
Like! But the XML library replacement sounds like possibly too high a cost.

**Comment by jenniferliddle** [comment]
I don't think so. I wouldn't be trying to write a generic XML parser, but a parser for known, specific XML config files. It shouldn't be too difficult.

**Comment by dkj** [comment]
Chatting to our colleagues [fopencookie](http://man7.org/linux/man-pages/man3/fopencookie.3.html) _might_ provide a generic way of obtaining standard file handles from [hfile](/samtools/htslib/blob/develop/htslib/hfile.h).

**Comment by jenniferliddle** [comment]
Ah Ha!
libXML does provide a xmlParseMemory() function that I can use in place of xmlReadFile(). Which works.
Ok, so I don't get to write my own XML parser. I can live with that :-)

**Comment by jenniferliddle** [comment]
I'd like to be able to test this against AWS S3 files.

> AWS offers the ‘IAM’ system; we can have one ‘root account’ (wtsi-npg) with multiple accounts attached (dj3, an8, jmtc, etc.), and all of these ‘subaccounts’ are billed via the root account, which makes billing easier.

@dkj could you please add the 'js10' subaccount to the 'wtsi-npg' root account?

**Comment by dkj** [comment]
Need to avoid `opendir` and `readdir` but as this is only used for instrument file format detection we can probably just try obtaining the different file names/formats for tile1, lane \<requested\>||1, cycle1 .
